### PR TITLE
218/jake/staking tab

### PR DESCRIPTION
**Pull request by jakegraves-figure**

### CODE REVIEW

File: lib/screens/home/explorer/explorer_bloc.dart [hunk 1]
```diff
@@ -1,32 +1,236 @@
...
+        _stakingDetails.value.selectedState,
+      );
+
+      final validators = await _validatorService.getRecentValidators(
+        _accountDetails.coin,
+        _validatorPages.value,
+        _stakingDetails.value.selectedStatus,
+      );
+
+      _stakingDetails.tryAdd(
```
Comment (royd) [inline lib/screens/home/explorer/explorer_bloc.dart:77]: 👍  `tryAdd`

File: lib/screens/home/explorer/explorer_bloc.dart [hunk 2]
```diff
@@ -1,32 +1,236 @@
...
+        StakingDetails(
+          abbreviatedValidators: _abbreviatedValidators,
+          delegates: delegations,
+          validators: validators,
+          address: account.address,
+        ),
+      );
+    } finally {
```
Comment (royd) [inline lib/screens/home/explorer/explorer_bloc.dart:85]: 👍  `finally`

File: lib/screens/home/explorer/explorer_bloc.dart [hunk 3]
```diff
@@ -1,32 +1,236 @@
...
+      ),
+    );
+  }
+
+  Future<List<T>> _loadMore<T>(
+      List<T> oldList,
+      BehaviorSubject<int> pages,
+      BehaviorSubject<bool> isLoading,
+      Future<List<T>> Function() function) async {
+    if (pages.value * 30 > oldList.length) {
```
Comment (royd) [inline lib/screens/home/explorer/explorer_bloc.dart:151]: `30` could be a named `const`

File: lib/screens/home/explorer/explorer_bloc.dart [hunk 4]
```diff
@@ -1,32 +1,236 @@
...
+    if (newList.isNotEmpty) {
+      oldList.addAll(newList);
+    }
+    return oldList;
+  }
+
+  Future<void> loadAdditionalDelegates() async {
+    var oldDetails = _stakingDetails.value;
+
+    final delegates = await _loadMore(
```
Comment (royd) [inline lib/screens/home/explorer/explorer_bloc.dart:167]: Might be worth creating a generic paged cache class that can work for this and other paged services, and write unit tests around it.

**Review by royd — APPROVED** [review]

File: lib/screens/home/explorer/explorer_bloc.dart (hunk 4 above)
Comment (jakegraves-figure) [reply lib/screens/home/explorer/explorer_bloc.dart:167]: It will take a bit of brainstorming, but I love it.

**Pull request merged**